go/vt/vttablet/tabletserver/vstreamer/rowstreamer.go

```diff
@@ -150,9 +150,14 @@ func (rs *rowStreamer) buildPlan() error {
 		return err
 	}
 	ti := &Table{
-		Name:   st.Name,
-		Fields: st.Fields,
+		Name: st.Name,
 	}
+
+	ti.Fields, err = getFields(rs.ctx, rs.cp, st.Name, rs.vschema.keyspace, st.Fields)
```

Comment:
Passing `rs.vschema.keyspace` may be problematic if the underlying database name is different from the keyspace name.

Reply:
Good catch. Sorry, I had caught it in the vstreamer call but forgot to update the rowstreamer one. Fixed here: https://github.com/vitessio/vitess/pull/13285/commits/700e2dd0835073330e9c26d231939420d10cbed6

Reply:
done in https://github.com/vitessio/vitess/pull/13045/commits/48fd6ee0b7ebf9e2f9a339bae303df55844f616a